helm-charts/splunk-otel-collector/templates/clusterRole.yaml

```diff
@@ -10,76 +10,7 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
 rules:
-{{- if eq (include "splunk-otel-collector.distribution" .) "openshift" }}
-- apiGroups:
-  - quota.openshift.io
-  resources:
-  - clusterresourcequotas
-  verbs:
-  - get
-  - list
-  - watch
-{{- end }}
-- apiGroups:
-  - ""
-  resources:
-  - events
-  - namespaces
-  - namespaces/status
-  - nodes
-  - nodes/spec
-  - nodes/stats
-  - nodes/proxy
-  - pods
-  - pods/status
-  - persistentvolumeclaims
-  - persistentvolumes
-  - replicationcontrollers
-  - replicationcontrollers/status
-  - resourcequotas
-  - services
-  verbs:
-  - get
-  - list
-  - watch
-- apiGroups:
-  - apps
-  resources:
-  - daemonsets
-  - deployments
-  - replicasets
-  - statefulsets
-  verbs:
-  - get
-  - list
-  - watch
-- apiGroups:
-  - extensions
-  resources:
-  - daemonsets
-  - deployments
-  - replicasets
-  verbs:
-  - get
-  - list
-  - watch
-- apiGroups:
-  - batch
-  resources:
-  - jobs
-  - cronjobs
-  verbs:
-  - get
-  - list
-  - watch
-- apiGroups:
-    - autoscaling
-  resources:
-    - horizontalpodautoscalers
-  verbs:
-    - get
-    - list
-    - watch
+{{- include "splunk-otel-collector.clusterRole.rules" . }}
```

Comment:
I think for readability and maintenance it's preferable to still have the rules in this file instead of migrating them to the values file which is already a little unwieldy.

```
Template files should have the extension .yaml if they produce YAML output.
The extension .tpl may be used for template files that produce no formatted content.
```
- https://helm.sh/docs/chart_best_practices/templates/

Reply:
Thanks for the advice! Firstly I adjusted this, but then I deleted it completely, according to the conversation below. Now user needs to set it by himself using `rbac.customRules`